huaweicloud/services/dew/resource_huaweicloud_kps_keypair.go

```diff
@@ -0,0 +1,358 @@
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(5 * time.Minute),
+			Delete: schema.DefaultTimeout(5 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.All(
+					validation.StringLenBetween(1, 64),
+					validation.StringMatch(regexp.MustCompile(`^[\-_A-Za-z0-9]+$`),
+						"The name can contain a maximum of 64 characters, including letters, digits, underscores (_)"+
+							" and hyphens (-)."),
+				),
+			},
+
+			"type": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Computed:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringInSlice([]string{"ssh", "x509"}, false),
+			},
+
+			"scope": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Computed:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringInSlice([]string{"domain", "user"}, false),
+			},
+
+			"encryption_type": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Computed:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringInSlice([]string{"default", "kms"}, false),
+			},
+
+			"encryption_kms_key_name": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+
+			"public_key": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+
+			"key_file": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+
+			"created_at": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"fingerprint": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"is_managed": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceKeypairCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	c := meta.(*config.Config)
+	region := c.GetRegion(d)
+	client, err := config.NewKmsClient(c, region)
+	if err != nil {
+		return fmtp.DiagErrorf("Error creating KMS v3 client: %s", err)
+	}
+
+	var importPublicKey *string
+	if v, ok := d.GetOk("public_key"); ok {
+		importPublicKey = utils.String(v.(string))
+	}
+
+	createOpts, err := buildCreateParams(d)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	logp.Printf("[DEBUG] Create KeyPair : %#v", createOpts)
+
+	response, err := client.CreateKeypair(createOpts)
+	if err != nil {
+		return fmtp.DiagErrorf("Error creating KeyPair: %s", err)
+	}
+
+	d.SetId(*response.Keypair.Name)
+
+	//update description
```

Comment:
please add a space between // and your comments

Reply:
done